test/e2e/tuning_options_test.go

```diff
@@ -320,3 +326,201 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
 		}
 	}
 }
+
+// TestCookieLen verifies that when logging is enabled and a cookie capture is specified, any captured cookies are
+// truncated at the expected max length.
+func TestCookieLen(t *testing.T) {
+	testCases := []struct {
+		name      string
+		maxLength int
+	}{
+		// 63 bytes was the limit before OCPBUGS-3356 was fixed; This should pass even without the patch, so it's more
+		// of a sanity check.
+		{"63 bytes", 63},
+		// The lower bound. 64 bytes is the smallest number that would be truncated short prior to the OCPBUGS-3356 fix.
+		{"64 bytes", 64},
+		// 128 bytes is just a number between the upper and lower bounds. If this passes and either boundary cases fail,
+		// it's likely an off-by-one or similar boundary issue.
+		{"128 bytes", 128},
+		// The upper bound. 1024 is the maximum allowed by the API.
+		{"1024 bytes", 1024},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			maxLength := testCase.maxLength
+			t.Parallel()
+			icName := types.NamespacedName{
+				Namespace: operatorNamespace,
+				Name:      names.SimpleNameGenerator.GenerateName("test-cookielen-"),
+			}
+			domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+			ic := newPrivateController(icName, domain)
+			cookieName := "X-foo-bar"
+			ic.Spec.Logging = &operatorv1.IngressControllerLogging{
+				Access: &operatorv1.AccessLogging{
+					Destination: operatorv1.LoggingDestination{
+						Type: "Container",
+						Container: &operatorv1.ContainerLoggingDestinationParameters{
+							MaxLength: 2048,
+						},
+					},
+					HTTPCaptureCookies: []operatorv1.IngressControllerCaptureHTTPCookie{{
+						IngressControllerCaptureHTTPCookieUnion: operatorv1.IngressControllerCaptureHTTPCookieUnion{
+							MatchType: operatorv1.CookieMatchTypeExact,
+							Name:      cookieName,
+						},
+						MaxLength: maxLength,
+					}},
+					HttpLogFormat: "cookie: %CC",
+				},
+			}
+			if err := kclient.Create(context.TODO(), ic); err != nil {
+				t.Fatalf("Failed to create ingresscontroller %s: %v", icName, err)
+			}
+			defer assertIngressControllerDeleted(t, kclient, ic)
+
+			if err := waitForIngressControllerCondition(t, kclient, 2*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+				t.Fatalf("Timed out waiting for ingresscontroller %s to become available: %v", icName.Name, err)
+			}
+
+			// Send a request through the router that includes the captured cookie, and verify that it's truncated as expected.
+
+			// Use the router image for the client pod since it includes curl.
+			routerDeployment := &appsv1.Deployment{}
```

Comment:
Could you take a quick look at how `buildCurlPod` is done in https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/operator_test.go#L3093 `TestUniqueIDHeaders`?  I think you should build your own simple image, route, and service rather than use the ones provided by router, canary and ingress.

Reply:
The advantage to using the canary route is that it's guaranteed to already be deployed, which speeds up the test by not requiring additional pods and routes to be created. That said, it does make the test more confusing for a pretty small gain. I'll rework this to use an echo pod like `TestUniqueIDHeaders`.